src/Symfony/Validator/Metadata/Property/Restriction/PropertySchemaCollectionRestriction.php

```diff
@@ -52,6 +52,10 @@ public function create(Constraint $constraint, ApiProperty $propertyMetadata): a
             }
 
             $restriction['properties'][$field] = $this->mergeConstraintRestrictions($baseConstraint, $propertyMetadata);
+
+            if ([] === $restriction['properties'][$field]) {
```

Comment:
I'm not a huge fan of this, I'm wonderign if we couldn't do better.

Reply:
What do you think about doing this instead  `return array_merge([], ...$propertyRestrictions) ?: new \ArrayObject()`  ?